### PR TITLE
fix: use rollup to output js files

### LINKED ISSUE #2913: 6.0 broke the build
**Marked version:**

6.0.0

**Describe the bug**
We've been using Marked as a CDN resource for multiple demo pages we create. With the 6.0 release, these pages stopped working. 

I am not sure if there is a bug or if there is a new config consideration that I am not seeing in the docs. 

```
Uncaught (in promise) ReferenceError: marked is not defined
```

**To Reproduce**
You can see how we use Marked in this example. 
https://github.com/AlaskaAirlines/auro-alert/blob/9188f2b4cbf0f319e3ce59bfa326611ecc3c0b5e/demo/index.html#L19-L29

Again, setting the version to 5.1.0, this code renders the markdown as expected. 

```js
    <script type="module">
      import 'https://unpkg.com/marked@5.1.0/marked.min.js';
      import 'https://unpkg.com/prismjs@latest/prism.js';
      fetch('/demo/demo.md')
        .then((response) => response.text())
        .then((text) => {
          const rawHtml = marked.parse(text);
          document.querySelector('main').innerHTML = rawHtml;
          Prism.highlightAll();
        });
    </script>
```

**Expected behavior**

The markdown renders as HTML as expected.

### PR DESCRIPTION
**Pull request by UziTech**
**Marked version:** v6.0.0

## Description

Fix umd output by removing tsup and adding rollup

Add umd-test to check umd adds global marked

- Fixes #2913 

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

**Review by calculuschild — APPROVED** [review]
Lgtm

**Comment by UziTech** [comment]
@styfle any issue with merging this?

File: tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es2021",
+    "target": "es2022",
```
Comment (styfle) [inline tsconfig.json:3]: I believe we chose ES2021 to support Node.js 16

File: package.json [hunk 2]
```diff
@@ -67,31 +68,32 @@
...
     "test:specs": "npm test -- test/specs/**/*-spec.js",
     "test:lint": "eslint .",
     "test:redos": "node test/recheck.js > vuln.js",
+    "test:types": "tsc --project tsconfig-type-test.json",
+    "test:umd": "node test/umd-test.js",
     "test:update": "node test/update-specs.js",
     "rules": "node test/rules.js",
     "bench": "npm run build && node test/bench.js",
     "lint": "eslint --fix .",
-    "type-check": "tsc",
```
Comment (styfle) [inline package.json:90]: Maybe I'm missing it, but where do we run `tsc` using the root `tsconfig.json`? Did you verify that a typo will fail the build?

File: tsconfig.json (hunk 1 above)
Comment (UziTech) [reply tsconfig.json:3]: Es 2021 doesn't support private methods that we already use. This still passes with node 16 with 2022.

File: package.json (hunk 2 above)
Comment (UziTech) [reply package.json:90]: Yes rollup handles running tsc and will fail to build on typescript error

**Review by styfle — APPROVED** [review]

**Pull request merged**